api/src/test/java/org/openmrs/module/fhir2/api/impl/FhirPatientServiceImplTest.java

```diff
@@ -873,6 +887,68 @@ public void getPatientEverything_shouldReturnAllInformationAboutAllPatients() {
 		assertThat(resultList.size(), greaterThanOrEqualTo(1));
 	}
 	
+	@Test
+	public void getPatientIdentifierTypeByIdentifier_shouldReturnIdentifierTypeWhenPresent() {
+		String typeName = "some-type";
+		
+		Identifier identifier = new Identifier();
+		identifier.setType(new CodeableConcept().setText(typeName));
+		
+		PatientIdentifierType expectedType = new PatientIdentifierType();
+		expectedType.setName(typeName);
+		
+		when(dao.getPatientIdentifierTypeByNameOrUuid(typeName, null)).thenReturn(expectedType);
+		
+		PatientIdentifierType result = patientService.getPatientIdentifierTypeByIdentifier(identifier);
+		assertNull(identifier.getSystem());
+		assertEquals(expectedType, result);
+	}
+	
+	@Test
+	public void getPatientIdentifierTypeByIdentifier_shouldReturnIdentifierTypeWithSystemWhenSystemIsPresent() {
+		String systemName = "some-system";
+		String expectedTypeName = "some-type";
+		String anotherTypeName = "another-type";
+		
+		Identifier identifierWithSystem = new Identifier();
+		identifierWithSystem.setSystem(systemName);
+		identifierWithSystem.setType(new CodeableConcept().setText(expectedTypeName));
+		
+		PatientIdentifierType expectedType = new PatientIdentifierType();
+		expectedType.setName(expectedTypeName);
+		
+		PatientIdentifierType unexpectedType = new PatientIdentifierType();
+		unexpectedType.setName(anotherTypeName);
+		
+		when(systemDao.getPatientIdentifierTypeByUrl(systemName)).thenReturn(expectedType);
+		
+		PatientIdentifierType result = patientService.getPatientIdentifierTypeByIdentifier(identifierWithSystem);
+		assertNotEquals(expectedType, unexpectedType);
+		assertEquals(expectedType, result);
+	}
+	
+	@Test
+	public void getPatientIdentifierTypeByIdentifier_shouldReturnNullWhenSystemAndIdentifierTypeNotPresent() {
+		Identifier identifier = new Identifier();
+		identifier.setType(null);
+		
+		PatientIdentifierType result = patientService.getPatientIdentifierTypeByIdentifier(identifier);
+		assertNull(identifier.getSystem());
+		assertFalse(identifier.hasType());
+		assertNull(result);
+	}
+	
+	@Test
+	public void getPatientIdentifierTypeByIdentifier_shouldReturnNullWhenIdentifierTypeTextEmpty() {
+		Identifier identifier = new Identifier();
+		identifier.setType(new CodeableConcept().setText(Strings.EMPTY));
```

Comment:
```suggestion
		identifier.setType(new CodeableConcept().setText(null));
```
any reason for not using null instead (as shown in the suggestion above)?

Reply:
Thanks @mherman22 for your suggestion. No reason for it, I was just trying to test the statement:

`StringUtils.isBlank(identifier.getType().getText()`

You have a good point that a null test would be good to check it is handled correctly too, so I have added a separate test for that now. I also realised there is no test for the case where the Identifier.type is null so also added a test case for this.